tests/tests/test_state_scripts.py

```diff
@@ -597,31 +601,28 @@ def test_reboot_recovery(
             MenderDeviceGroup([mender_device.host_string])
         )[mender_device.host_string]
 
-        host_ip = (
-            class_persistent_standard_setup_one_client_bootstrapped.get_virtual_network_host_ip()
-        )
+        host_ip = setup_client_state_scripts_update_module.get_virtual_network_host_ip()
+
+        def make_artifact(filename, artifact_name):
+            return image.make_module_artifact(
+                "module-state-scripts-test",
+                conftest.machine_name,
+                artifact_name,
+                filename,
+                scripts=[artifact_script_dir],
+            )
+
         with mender_device.get_reboot_detector(host_ip) as reboot_detector:
 
             common_update_procedure(
-                install_image=new_rootfs,
                 verify_status=True,
                 devices=[device_id],
                 scripts=[artifact_script_dir],
-            )[0]
+                make_artifact=make_artifact,
+            )
 
             try:
-
-                orig_part = mender_device.get_active_partition()
-
-                # handle case where the client has not finished the update
-                # path on the committed partition, but new partition is installed,
-                # thus we will not get a valid entrypoint into the uncommitted parition(reboot_leave)
-                # and the client will thus reboot straight after starting, and u-boot will
-                # fall back to the committed partition
-                if test_set.get("DoubleReboot", False):
-                    reboot_detector.verify_reboot_performed(number_of_reboots=2)
-                else:
-                    reboot_detector.verify_reboot_performed()
+                reboot_detector.verify_reboot_performed()
```

Comment:
I'm not quite clear on why this is still here? This only detects real reboots, not `systemctl restart` reboots, AFAIK.

Reply:
Yes, in only detects that the spontaneous reboot (via `sysrq-trigger`) happened. Note that such check is completely removed on `test_state_scripts` below. So it is rather a validation of the test logic than the device behavior.

Do you suggest removing it completely, then?

Reply:
Oh, right, it is needed for the spontaneous reboots. No, then you're right, it should stay. It must stay actually, because otherwise you risk just reconnecting to the previous boot's ssh (this is why we introduced the reboot detector).